markdownlint/TOP005_blanksAroundMultilineHtmlTags/TOP005_blanksAroundMultilineHtmlTags.js

```diff
@@ -0,0 +1,76 @@
+module.exports = {
+  names: ["TOP005", "blanks-around-multiline-html-tags"],
+  description: "Multiline HTML tags should be surrounded by blank lines or code block delimiters",
+  tags: ["html", "blanks"],
+  information: new URL(
+    "https://github.com/TheOdinProject/curriculum/blob/main/markdownlint/docs/TOP005.md"
+  ),
+  function: function TOP005(params, onError) {
+    /**
+     * HTML code in HTML/JSX code blocks should not be flagged.
+     * We only want to flag HTML tags we use for actual markup,
+     * or md code block examples of such.
+     */
+    const IGNORED_FENCE_TYPES = ["html", "jsx"];
+    const ignoredFencesLineNumbers = params.tokens
+      .filter((token) => {
+        return token.type === "fence" && IGNORED_FENCE_TYPES.includes(token.info);
+      })
+      .map((token) => [token.map[0], token.map[1] - 1]); // 0-indexing
+
+    const isWithinIgnoredFence = (lineNumber) => {
+      return ignoredFencesLineNumbers.some(
+        (range) => range[0] < lineNumber && lineNumber < range[1]
+      );
+    };
+
+    /**
+     * html_block tokens are only separated once you hit a blank line.
+     *
+     * <div>
+     * foo
+     * </div>
+     * <div>
+     * bar
+     * </div>
+     *
+     * gets treated as a single html_block with the contents:
+     * `<div>\nfoo\n</div>\n<div>\nbar\n</div>\n`
+     * which will require splitting.
+     *
+     * Therefore, manual matching is more straightforward than relying on tokens.
+     */
```

Comment:
Not really sure we need this comment. I think the README would explain the issue with no blank lines/what kinda happens internally with markdown-it.